src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -468,6 +486,30 @@ public static ModuleFinder newModuleFinder(List<Path> paths,
         return finder;
     }
 
+    private static void deleteDirectory(Path dir) throws IOException {
+        if (dir != null && Files.isDirectory(dir)) {
```

Comment:
I don't think this method should be checking dir, that should be up to the caller to ensure that it is not called when output is null.

Reply:
okay. will move that check to caller.